compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5962,6 +5963,165 @@ OMR::X86::TreeEvaluator::bitpermuteEvaluator(TR::Node *node, TR::CodeGenerator *
    return resultReg;
    }
 
+TR::Register*
+OMR::X86::TreeEvaluator::compressbitsEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::DataType dt = node->getDataType();
+
+   TR_ASSERT_FATAL(dt == TR::Int32 || dt == TR::Int16 || dt == TR::Int8, "compressbitsEvaluator only supports 32, 16, or 8 bits");
+
+   TR::Node *srcNode = node->getFirstChild();
+   TR::Node *maskNode = node->getSecondChild();
+   TR::Register *srcReg = NULL;
+
+   if (dt != TR::Int32)
+      {
+      // src needs to be zero-extended to 32 bits
```

Comment:
I've been puzzling about this.  Is there a case where zero-extending the source would yield a different result from zero-extending the mask for `{b|s}compressbits`?  If so, I think it would be worthwhile adding a comment giving examples.

If not, I think you would be able to change this to zero-extend the mask and share code between `compressbitsEvaluator` and `expandbitsEvaluator` in the same way that you did with `lcompressbitsEvaluator` and `lexpandbitsEvaluator`.

Reply:
I revisited the need for zero extension and I have a couple observations to modify my implementations here:
- zero extension is unnecessary for `expandbits`
- either zero extending the source or mask will give the same result for `compressbits`

After adding these modifications `compressbitsEvaluator` and `expandbitsEvaluator` will be more different and might be better left separate.